### ISSUE TITLE
First time loading videos sometimes takes a while to load (Needs more testing)

### ISSUE THREAD

**Issue by trayfellow** [polybiusproxy/hxCodec#286]
With hxCodec 3.0 loading videos for the first time can take to 10-15 seconds to load

**Comment by NeeEoo** [comment]
Can confirm this also happens for me on mac, and like other people i see playing a mod i worked on

**Comment by JonnycatMeow** [comment]
i have an idea

**Comment by JonnycatMeow** [comment]
we can use file caching

**Comment by JonnycatMeow** [comment]
![Screen Shot 2023-07-15 at 5 59 20 PM](https://github.com/polybiusproxy/hxCodec/assets/107599365/330ed368-ff7a-486d-8681-dfc3a0e25755)

**Comment by EgguMan** [comment]
This occurs because the video is being moved from hard storage to RAM, and videos are very dense.

**Comment by JonnycatMeow** [comment]
is there anyway to fix it?

**Comment by EgguMan** [comment]
> is there anyway to fix it?

The way I did it is with this little work around on a seperate thread in my loading state.

![image](https://github.com/polybiusproxy/hxCodec/assets/100107293/38246268-8545-47e0-825c-d334df5ce912)

**Comment by JonnycatMeow** [comment]
well could there be another way for it to be a perminate fix for hxCodec

**Comment by EgguMan** [comment]
> well could there be another way for it to be a perminate fix for hxCodec

To my understanding no. To do that it would need to preload every single video in the game at once. Not only is that extremely stressful on the PC but it's very resource dependent

**Comment by JonnycatMeow** [comment]
oh ok

**Comment by JonnycatMeow** [comment]
well can you make a example on the readme and name it preload video so people can do what you did to the video

**Comment by EgguMan** [comment]
> well can you make a example on the readme and name it preload video so people can do what you did to the video 

Nah, it all works off a custom cutscene framework I made for my mod

**Comment by JonnycatMeow** [comment]
oh

**Comment by JonnycatMeow** [comment]
what is your mod called and does it have a source code on github?

**Comment by polybiusproxy** [comment]
Hello, does this issue occur when loading small videos? How big are the videos you're trying to load?

**Comment by EgguMan** [comment]
> Hello, does this issue occur when loading small videos? How big are the videos you're trying to load?

it appears to be based on the size of the videos, from my limited testing

**Comment by Karim-AkraNo2FA** [comment]
![IMG-20231022-WA0001(1)](https://github.com/polybiusproxy/hxCodec/assets/144803230/071159e0-7cbd-4ac1-9938-86810b916127)
/

I manged to make it load faster using sys.threads.Thread

**Comment by polybiusproxy** [comment]
Won't fix atm, it is expected for hxCodec to take time when loading big videos, due to it being single-threaded, slow I/O speed on the hardware side, etc.

**Comment by JonnycatMeow** [comment]
hxvlc has faster loading time and they already did the byte video support

**Comment by JonnycatMeow** [comment]
well i am gonna try to make vlc wasm
